pkg/api/v1alpha1/cluster_types.go

```diff
@@ -440,9 +442,45 @@ func WorkerNodeGroupConfigurationsSliceEqual(a, b []WorkerNodeGroupConfiguration
 		return false
 	}
 
+	if !WorkerNodeGroupConfigurationsKubeVersionEqual(a, b) {
```

Comment:
why are we repeating the whole map thing? Can't we just extend the algorithm above this to take this version into account?